src/kernels/webgl/glsl_version.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {ENV} from '../../environment';
+
+export type GLSL = {
+  version: string,
+  attribute: string,
+  varyingVs: string,
+  varyingFs: string,
+  texture2D: string,
+  output: string,
+  defineOutput: string,
+  defineRound: string
+};
+
+export function getGlslDifferences(): GLSL {
+  let version: string;
+  let attribute: string;
+  let varyingVs: string;
+  let varyingFs: string;
+  let texture2D: string;
+  let output: string;
+  let defineOutput: string;
+  let defineRound: string;
+
+  if (ENV.get('WEBGL_VERSION') === 2) {
+    version = '#version 300 es';
+    attribute = 'in';
+    varyingVs = 'out';
+    varyingFs = 'in';
+    texture2D = 'texture';
+    output = 'outputColor';
+    defineOutput = 'out vec4 outputColor;';
+    defineRound = '#define round(value) int(floor((value) + 0.5))';
```

Comment:
Not use `'#define round(value) int(round(value))'` here because the built-in round just return the nearest integer. https://www.khronos.org/registry/OpenGL-Refpages/gl4/